src/stash/src/transaction.rs

```diff
@@ -68,6 +74,16 @@ pub struct Transaction<'a> {
     // Savepoint state to enforce the invariant that only one SAVEPOINT is
     // active at once.
     savepoint: Arc<Mutex<bool>>,
+
+    // Cached sinces and uppers for this transaction. These are set on first
+    // query and updated on seal/compact.
+    sinces: Arc<Mutex<HashMap<Id, Antichain<Timestamp>>>>,
```

Comment:
@guswynn Should these be tokio mutexes? I'm not sure what happens to a tokio task when a std Mutex lock blocks and waits.